### PR TITLE
langchain: Documentation fix for #16702 and #16703

### PR DESCRIPTION
**Pull request by Naveenkhasyap**
- **Description:** Quickstart Documentation updates for missing dependency installation steps.
  - **Issue:** the issue # it prompts users to install required dependency.
  - **Dependencies:** no,
  - **Twitter handle:** @naveenkashyap_

### CODE REVIEW

File: docs/docs/get_started/quickstart.mdx [hunk 1]
```diff
@@ -471,6 +472,7 @@ While the first part of this guide was intended to be run in a Jupyter Notebook,
 Install with:
 ```bash
 pip install "langserve[all]"
+pip install "fastapi"
```
Comment (hwchase17) [inline docs/docs/get_started/quickstart.mdx:475]: this is installed via the above installation

File: docs/docs/get_started/quickstart.mdx [hunk 2]
```diff
@@ -417,6 +417,7 @@ Now that we have the tools, we can create an agent to use them. We will go over
 Install langchain hub first
 ```bash
 pip install langchainhub
+pip install langchain_openai
```
Comment (hwchase17) [inline docs/docs/get_started/quickstart.mdx:420]: lets not lump this in with langchainhub but include a specific instruction on why were installing it
also, if you are following this guide, theres a good chance youve already installed it

File: docs/docs/get_started/quickstart.mdx (hunk 1 above)
Comment (Naveenkhasyap) [reply docs/docs/get_started/quickstart.mdx:475]: got error package not found after installing **langserve[all]** when following the guides 
platform: Mac Pro - M2 chip
Python : 3.9
Pip: 23.3.2

File: docs/docs/get_started/quickstart.mdx (hunk 2 above)
Comment (Naveenkhasyap) [reply docs/docs/get_started/quickstart.mdx:420]: sure we can add one more paragraph explaining why we need this

Comment (Naveenkhasyap) [reply docs/docs/get_started/quickstart.mdx]: updated

File: docs/docs/get_started/quickstart.mdx (hunk 1 above)
Comment (Naveenkhasyap) [reply docs/docs/get_started/quickstart.mdx]: removed this since it has been added in langserve[all] dependency list

**Review by hwchase17 — APPROVED** [review]

File: docs/docs/get_started/quickstart.mdx [hunk 3]
```diff
@@ -427,7 +432,7 @@ from langchain.agents import create_openai_functions_agent
 from langchain.agents import AgentExecutor
 
 # Get the prompt to use - you can modify this!
-prompt = hub.pull("hwchase17/openai-functions-agent")
+# you need to set OPENAI_API_KEY envirnment variable to use this or pass it as argument as follows *openai_api_key=sk-...*
```
Comment (hwchase17) [inline docs/docs/get_started/quickstart.mdx:435]: wait - we still need the prompt here

**Review by hwchase17 — DISMISSED** [review]

**Review by baskaryan — APPROVED** [review]

**Pull request merged**